submissions/smillims/oop-exercise/index.js

```diff
@@ -0,0 +1,85 @@
+/* 
+  Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+*/
+
+class Inhabitant {
+  constructor(surname, species, gender, name, legs){
+    this.surname = surname;
+    this.species = species;
+    this.gender = gender;
+    this.name = name;
+    this.legs = legs;
+  }
+  toString(){
+    return [`Hi, I'm ` + this.species,
+    `My name is ` + this.name,
+    `My gender is ` + this.gender,
+    `I have ` + this.legs + ` legs`, ``].join(', ')
+  }
```

Comment:
@smillims Generalization is a concept a bit wider than just making some shared prop a property of a base class.
Any good reasons to handle `surname` in a way different than e.g. `name`?

Reply:
No reason, my fail)